src/EvaluateEquationDerivative.jl

```diff
@@ -259,63 +241,70 @@ function eval_grad_tree_array(
     )
 end
 
-function _eval_grad_tree_array(
+@generated function _eval_grad_tree_array(
     tree::AbstractExpressionNode{T},
-    ::Val{n_gradients},
+    n_gradients,
     index_tree::Union{NodeIndex,Nothing},
     cX::AbstractMatrix{T},
     operators::OperatorEnum,
     ::Val{variable},
-    ::Val{turbo},
-)::Tuple{
-    AbstractVector{T},AbstractMatrix{T},Bool
-} where {T<:Number,variable,turbo,n_gradients}
-    if tree.degree == 0
-        grad_deg0_eval(tree, Val(n_gradients), index_tree, cX, Val(variable))
-    elseif tree.degree == 1
-        grad_deg1_eval(
-            tree,
-            Val(n_gradients),
-            index_tree,
-            cX,
-            operators.unaops[tree.op],
-            operators.diff_unaops[tree.op],
-            operators,
-            Val(variable),
-            Val(turbo),
-        )
-    else
-        grad_deg2_eval(
-            tree,
-            Val(n_gradients),
-            index_tree,
-            cX,
-            operators.binops[tree.op],
-            operators.diff_binops[tree.op],
-            operators,
-            Val(variable),
-            Val(turbo),
-        )
+)::ResultOk2 where {T<:Number,variable}
+    nuna = get_nuna(operators)
+    nbin = get_nbin(operators)
+    quote
+        if tree.degree == 0
+            grad_deg0_eval(tree, n_gradients, index_tree, cX, Val(variable))
+        elseif tree.degree == 1
+            op_idx = tree.op
+            Base.Cartesian.@nif(
+                $nuna,
+                i -> i == op_idx,
+                i -> grad_deg1_eval(
+                    tree,
+                    n_gradients,
+                    index_tree,
+                    cX,
+                    operators.unaops[i],
+                    operators,
```

Comment:
Actually we can probably undo this bit since we aren't using Enzyme on Zygote code anyways. Maybe if there's a need we can add it back.